### DISCUSSION TITLE
Is there a way to synthesize a spectrum for specific x axis values?

### DISCUSSION THREAD

**Discussion by nicocopez** [h2020charisma/ramanchada2#73] [Q&A] [answered]
I want to synthesize a spectrum which has a given x-axis range. Normally, if I get the x-axis vector of a spectrum, the space between the x-axis points are not equal. Is there a straight forward way to do so within ramanchada2?

I see there is the xcal option, but could I used it to load a given x-axis vector?

Thank you in advance for your time!

**Comment by georgievgeorgi** [comment, marked as answer]
Yes, here is the example:

```python
import numpy as np
import ramanchada2 as rc2

xarr = np.array([1, 1.1, 2, 2.2, 3, 3.3, 4, 4.4, 5, 5.5])
yarr = np.random.exponential(size=len(xarr))
rc2.spectrum.Spectrum(x=xarr, y=yarr).plot(fmt=':.')
```

**Reply by nicocopez** [reply]
Ok, this is great! So if I got it right, I can create my spectrum with numpy functions and then convert it to the class Spectrum, correct? Thank you Georgi!

**Reply by georgievgeorgi** [reply]
`rc2.spectrum.Spectrum(x, y, metadata)` expects `x` and `y` to be of type numpy array and returns a spectrum with these axes.
At the moment there is no strict check whether
```python
len(x) == len(y)
```
So the user should take care of that.
also note that in the resulting `Spectrum` x-axis will be sorted in ascending order so the following example will produce a straight line:
```python
xarr = np.array([10, 5, 1, 20])
rc2.spectrum.Spectrum(x=xarr, y=xarr).plot(fmt=':.')
```
